src/context/DataContext.tsx

```diff
@@ -0,0 +1,34 @@
+import React, { createContext, useState, ReactNode } from "react";
+
+interface Data {
+  name: string;
+  edad: string;
+}
+
+interface DataContextProps {
+  data: Data;
+  setData: React.Dispatch<React.SetStateAction<Data>>;
+}
+
+const dataFixed: Data = {
+  name: "Gabriel",
+  edad: "22",
+};
+
+interface DataProviderProps {
+  children: ReactNode;
+}
+
+export const DataContext = createContext<DataContextProps | undefined>(
+  undefined
+);
+
+const DataProvider = ({ children }: DataProviderProps) => {
+  const [data, setData] = useState<Data>(dataFixed);
+  return (
+    <DataContext.Provider value={{ data, setData }}>
```

Comment:
Instead of exposing the entire dispatcher function, I recommend creating a function that executes the dispatcher function right here, inside the DataProvider. This will also simplify the Type on line 10.

```
const updateData = (data: Data) => {
setData(data)
}
```

and the type now is:

```
interface DataContextProps {
  data: Data;
 updateData: (data: Data) => void;
}
```

```suggestion
    <DataContext.Provider value={{ data,updateData }}>
```

😄